x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -365,6 +385,12 @@ public Block eval(Page page) {
                                 if (b.isNull(0)) {
                                     continue;
                                 }
+                                if (b.getValueCount(0) > 1) {
+                                    condition.conditionWarnings.registerException(
+                                        new IllegalArgumentException("CASE expects a single-valued boolean")
+                                    );
+                                    continue;
+                                }
```

Comment:
Here's the core of the fix! It's fun how it's like 5 lines of fix, 200 testing, 200 ceremony to plumb the fix properly, and another 200 of docs updates.